src/var/mutation_counting.jl

```diff
@@ -0,0 +1,152 @@
+# phylo/mutation_counting.jl
+# ==================
+#
+# Types and methods for counting different kinds of mutations between two
+# DNA or RNA sequences.
+#
+# Part of the Bio.Phylo module.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+# Counting mutations
+# ------------------
+
+# Mutation types
+
+immutable Different end
+immutable Transition end
+immutable Transversion end
+
+
+"""
+    is_ambiguous_strict{T<:Nucleotide}(a::T, b::T)
+
+The strictest test of ambiguity, if either nucleotide is not
+A, T, G, or C, then this function returns true.
+"""
+@inline function is_ambiguous_strict{T<:Nucleotide}(a::T, b::T)
+    return isambiguous(a) || isambiguous(b)
+end
+
+typealias alphabets Union{DNAAlphabet{2}, DNAAlphabet{4}, RNAAlphabet{2}, RNAAlphabet{4}}
+
+"""
+    count_mutations(seq1::BioSequence{A}, seq2::BioSequence{A}, kind::Type{Different})
+
+Count the number of differences between two DNA sequences.
+Returns a tuple of the number of sites that are different, and the number of sites
+considered (sites with any ambiguity characters are not considered).
```

Comment:
Insert a blank line after the first sentence.

> Include a single one-line sentence describing what the function does or what the object represents after the simplified signature block. If needed, provide more details in a second paragraph, after a blank line.
<http://julia.readthedocs.io/en/latest/manual/documentation/>